test/unit_tests/braket/experimental/autoqasm/test_pulse_control.py

```diff
@@ -0,0 +1,166 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""Tests for the pulse control module."""
+
+import textwrap
+
+import pytest
+
+import braket.experimental.autoqasm as aq
+from braket.experimental.autoqasm.instructions import (
+    barrier,
+    capture_v0,
+    delay,
+    h,
+    play,
+    set_frequency,
+    set_phase,
+    set_scale,
+    shift_frequency,
+    shift_phase,
+)
+from braket.pulse import ArbitraryWaveform, Frame, Port
+
+PORT = Port(port_id="device_port_x0", dt=1e-9, properties={})
+FRAME1 = Frame(frame_id="predefined_frame_1", frequency=2e9, port=PORT, phase=0, is_predefined=True)
+FRAME2 = Frame(frame_id="predefined_frame_2", frequency=1e6, port=PORT, is_predefined=True)
```

Comment:
This is something that we can hopefully improve for the AutoQASM experience relative to what exists. It's always been really tricky to know what frames or ports to select

Reply:
Agree! working in progress